compiler-core/src/exhaustiveness.rs

```diff
@@ -814,6 +816,29 @@ impl BranchMode {
     }
 }
 
+#[derive(Eq, PartialEq)]
+enum BranchingFactor {
+    Infinite,
+    Finite(usize),
+}
+
+impl PartialOrd for BranchingFactor {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for BranchingFactor {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        match (self, other) {
+            (BranchingFactor::Infinite, BranchingFactor::Infinite) => std::cmp::Ordering::Equal,
+            (BranchingFactor::Infinite, BranchingFactor::Finite(_)) => std::cmp::Ordering::Greater,
+            (BranchingFactor::Finite(_), BranchingFactor::Infinite) => std::cmp::Ordering::Less,
+            (BranchingFactor::Finite(one), BranchingFactor::Finite(other)) => one.cmp(other),
+        }
+    }
+}
```

Comment:
This is quite a bit harder for me to understand now, and it means we have to use traits. Please revert to using numbers. 🙏

Reply:
Done!